### ISSUE TITLE
The problem of PV usage less than 1G 

### ISSUE THREAD

**Issue by loginkeithhe** [democratic-csi/democratic-csi#137]
When a PV (10G allocated) actually occupies a capacity of less than 1G (for example, only 100M is used), we take a snapshot of it, and then create a new PV based on this snapshot, it will not succeed. It was found that the capacity limit of dataset should not be less than 1G on truenas.I see that the space size of the dataset is set to the size of PV 100M

If the actual use of PV space exceeds 1G, there is no problem.

**Comment by travisghansen** [comment]
Ah good catch! This is for nfs I suppose?

**Comment by travisghansen** [comment]
What is the size you are requesting for the new PV and how is that size generated?

**Comment by loginkeithhe** [comment]
Yes, this problem occurs in NFS storage.
I conducted a test and I used the KubeSphere integrated environment. When I created it from a snapshot with WebUI, it didn't prompt me how much space I needed (because it is set to PV actual used space by default). Of course, I can change it to create from a yaml file and set a space greater than or equal to 1Gi.
I think the problem should be that TrueNas does not allow dataset smaller than 1G to be set. 

Example:

`apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  namespace: demo-pm
  name: create-pv-baseon-snap
  labels: {}
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 29172Ki           ###The use space is set to be greater than 1Gi 
  dataSource:
    name: test-pv-snap
    kind: VolumeSnapshot
    apiGroup: snapshot.storage.k8s.io
  storageClassName: freenas-nfs-csi`
